src/domain/wallets/tx-history.ts

```diff
@@ -17,23 +17,26 @@ const filterPendingIncoming = (
         walletTransactions.push({
           id: rawTx.txHash,
           walletId,
-          initiationVia: PaymentInitiationMethod.OnChain,
-          settlementVia: SettlementMethod.OnChain,
+          settlementAmount: sats,
+          settlementFee: toSats(0),
+          settlementUsdPerSat: usdPerSat,
+          status: TxStatus.Pending,
+          memo: null,
+          createdAt: createdAt,
           deprecated: {
             description: "pending",
             usd: usdPerSat * sats,
             feeUsd: 0,
             type: LedgerTransactionType.OnchainReceipt,
           },
-          otherPartyUsername: null,
-          settlementFee: toSats(0),
-          transactionHash: rawTx.txHash,
-          status: TxStatus.Pending,
-          memo: null,
-          createdAt: createdAt,
-          settlementAmount: sats,
-          settlementUsdPerSat: usdPerSat,
-          address,
+          initiationVia: {
```

Comment:
if this is `initiationVia` without an appendix telling which protocol we are using, I would assume this would cover all the initiation option, yet we have only have `OnChain` in the object below?

Reply:
We differentiate this via the `type` sub-property inside `initiationVia`, not the actual property name. In this case though, I could actually make this more accurate by changing the return type of the `filterPendingIncoming` to be `WalletOnChainTransaction`.